### ISSUE TITLE
Improper array size error in identifysupportregions 

### ISSUE THREAD

**Issue by Intech1** [drlukeparry/pyslm#39]
While attempting to generate support material for a part, I receive the following error. It seems to be an array size mismatch, but I am unsure of the current cause.

<img width="607" alt="Capture" src="https://github.com/drlukeparry/pyslm/assets/147294449/aadf8265-0546-4b32-8566-e3bc64b4b7e9">

**Comment by drlukeparry** [comment]
Thank you for sharing the issue. 

This is known problem that can happen when some of the settings are not finally tuned.  Normally, the identified flattened  support region is processed into a polygon and a mesh is generated. Each point in this mesh is subsequently projected back onto the original mesh to generate the support block volumes.

However, If the simplification factor (`splineSimplicationFactor`) and the offset factors too large (`innerSupportEdgeGap`, `outerSupportEdgeGap`), these points may not coincidentally not project or intersect back with the mesh. 

I would suggest the following in this case:
- reduce the `rayProjectionResolution` to a lower value (e.g. 0.1 ) to capture the overhang boundaries regions more precisely
- reduce `splineSimplicationFactor` or set to 0 to eliminate over simplification of the boundary
- Adjust and increase the offset values from the support boundary for both `innerSupportEdgeGap`, and outerSupportEdgeGap

**Comment by Intech1** [comment]
You're welcome!! 
Thank you for the suggestion! I took your advice into consideration and made some adjustments to the code based on your recommendations but facing the same issue.
I have attached the code file and a portion of the STL file for your reference. I would greatly appreciate it if you could kindly assist me in resolving this issue. Your expertise and guidance would be invaluable in helping me overcome this challenge.
Thank you once again for your support and willingness to assist.

[support_structure.zip](https://github.com/drlukeparry/pyslm/files/12843062/support_structure.zip)

**Comment by drlukeparry** [comment]
Thank you for kindly sharing the example, which is very helpful to diagnose the issue further. 

I'll take a look and see what geometry related issue can be observed.

**Comment by drlukeparry** [comment]
I looked over this in the development release (dev branch), which has further improvements and checks regarding this similar issue and found that it could generate the lattice with your support parameters. 

<img width="598" alt="image" src="https://github.com/drlukeparry/pyslm/assets/59477422/227108e9-9219-4f0a-b315-2c94ec061b10">

It could be that I introduced a method to catch some of errors related to this issue. Presumably you are using the version 0.5 of PySLM?
